### PR TITLE
Documentation updates

### LINKED ISSUE #5: Documentation
Move and improve the documentation from stm32-rs.

### PR DESCRIPTION
**Pull request by jessebraham**
Closes #5. I basically just pulled in the **Device and Peripheral YAML Format** section along with its subsections from stm32-rs, as well as did some tidying up and reworking of the existing README.

If there are any further changes you would like made prior to merging just let me know and I'll be happy to make them.

### CODE REVIEW

**Comment by nickray** [comment]
Hey, thanks a lot - unexpected!

File: README.md [hunk 1]
```diff
@@ -2,36 +2,290 @@
...
+
+- Enumerated values should be named in the past tense ("enabled", "masked",
+  etc).
+- Descriptions should start with capital letters but do not end with a period
+
+
 ## License
 
-svdtools is licensed under either of
+Licensed under either of
```
[the remark this answers is not in the document]
Comment (jessebraham) [reply README.md]: You bet, just got a little copy-paste happy from the stm32-rs README 😄

**Comment by nickray** [comment]
Thanks again!

**Comment by jessebraham** [comment]
Glad to help!

**Pull request merged**